lib/pbench/test/unit/server/test_user_management_cli.py

```diff
@@ -0,0 +1,227 @@
+                TestUserManagement.EMAIL_SWITCH,
+                TestUserManagement.EMAIL_TEXT,
+                TestUserManagement.FIRST_NAME_SWITCH,
+                TestUserManagement.FIRST_NAME_TEXT,
+                TestUserManagement.LAST_NAME_SWITCH,
+                TestUserManagement.LAST_NAME_TEXT,
+                TestUserManagement.ROLE_SWITCH,
+                "ADMIN",
+            ],
+        )
+        assert result.exit_code == 0, result.stderr
+        assert result.stdout == "Admin user test_user registered\n"
+
+    @staticmethod
+    def test_user_creation_with_invalid_role(server_config, pytestconfig):
+        runner = CliRunner(mix_stderr=False)
+        result = runner.invoke(
+            cli.user_create,
+            args=[
+                TestUserManagement.USER_SWITCH,
+                TestUserManagement.USER_TEXT,
+                TestUserManagement.PSWD_SWITCH,
+                TestUserManagement.PSWD_TEXT,
+                TestUserManagement.EMAIL_SWITCH,
+                TestUserManagement.EMAIL_TEXT,
+                TestUserManagement.FIRST_NAME_SWITCH,
+                TestUserManagement.FIRST_NAME_TEXT,
+                TestUserManagement.LAST_NAME_SWITCH,
+                TestUserManagement.LAST_NAME_TEXT,
+                TestUserManagement.ROLE_SWITCH,
+                "ADMN",
+            ],
+        )
+        assert result.exit_code == 2, result.stderr
+        assert (
+            result.stderr.find(
+                "Invalid value for '--role': invalid choice: ADMN. (choose from ADMIN)"
+            )
+            > -1
+        )
+
+    @staticmethod
+    def test_valid_user_delete(monkeypatch, server_config, pytestconfig):
+        monkeypatch.setattr(User, "delete", mock_valid_delete)
+        runner = CliRunner(mix_stderr=False)
+
+        result = runner.invoke(cli.user_delete, args=[TestUserManagement.USER_TEXT])
+        assert result.exit_code == 0, result.stderr
+
+    @staticmethod
+    def test_invalid_user_delete(pytestconfig, server_config):
+        runner = CliRunner(mix_stderr=False)
+        # Give username that doesn't exists to delete
+        result = runner.invoke(cli.user_delete, args=["wrong_username"])
+        assert result.exit_code == 1
+        assert result.stderr == "User wrong_username does not exist\n"
+
+    @staticmethod
+    def test_valid_user_list(monkeypatch, server_config, pytestconfig):
+        monkeypatch.setattr(User, "query_all", mock_valid_list)
+        runner = CliRunner(mix_stderr=False)
+
+        result = runner.invoke(cli.user_list,)
+        assert result.exit_code == 0, result.stderr
+        assert (
+            result.stdout
+            == cli.USER_LIST_HEADER_ROW
+            + "\n"
+            + cli.USER_LIST_ROW_FORMAT.format(
+                TestUserManagement.USER_TEXT,
+                TestUserManagement.FIRST_NAME_TEXT,
+                TestUserManagement.LAST_NAME_TEXT,
+                datetime.datetime.now().strftime("%Y-%m-%d"),
```

Comment:
Hmm; there's actually a chance this will fail, if the tests run around midnight. You should use the timestamp generated when the mock user was created; or, sooner or later, someone's going to get an unpleasant present from GitHub. (Or, alternately, look for a YYYY-MM-DD pattern and don't worry about the actual date, which I think in this context would probably be fine.)

Reply:
The mock user gets created right above this assertion so even if there is a delay it is going to be very minuscule but okay I can look into setting a fixed timestamp on user

Reply:
It's easy enough to set a `TestUserManagement.REGISTERED_ON` value which is used in place of `datetime.date.now()` in the creation and checked in the assertion (it could even be initialized to `datetime.date.now()` -- the key is not that it is static but that it's the same value referenced in both places).

This is a common pattern in unit tests:  unit tests often communicate with the mocks to ensure that they are called and that the values they produce are used appropriately by the CUT.

Reply:
fixed